test/contracts-network/colony.js

```diff
@@ -419,22 +430,87 @@ contract("Colony", (accounts) => {
   });
 
   describe("when viewing deprecated Tasks and Payments", () => {
-    let tasksPayments;
+    let OldInterface;
+    let oldColony;
+    before(async () => {
+      await deployOldColonyVersion(
+        "Colony",
+        "IMetaColony",
+        [
+          // eslint-disable-next-line max-len
+          "Colony,ColonyDomains,ColonyExpenditure,ColonyFunding,ColonyPayment,ColonyRewards,ColonyRoles,ColonyTask,ContractRecovery,ColonyArbitraryTransaction",
+        ],
+        "glwss4",
+        colonyNetwork,
+      );
 
-    beforeEach(async () => {
-      const colonyVersion = await colony.version();
-      const colonyResolverAddress = await colonyNetwork.getColonyVersionResolver(colonyVersion);
-      const colonyResolver = await Resolver.at(colonyResolverAddress);
+      const oldFunctions = new ethers.utils.Interface([
+        "function makeTask(uint256,uint256,bytes32,uint256,uint256,uint256)",
+        "function addPayment(uint256, uint256, address, address, uint256, uint256, uint256)",
+      ]);
+      const formattedOldABI = JSON.parse(oldFunctions.format(ethers.utils.FormatTypes.json));
+      console.log(JSON.stringify(formattedOldABI));
+
+      for (let i = 0; i < formattedOldABI.length; i += 1) {
+        for (let j = 0; j < formattedOldABI[i].inputs.length; j += 1) {
+          formattedOldABI[i].inputs[j].internalType = formattedOldABI[i].inputs[j].type;
+          formattedOldABI[i].inputs[j].name = "";
+        }
+      }
+
+      OldInterface = truffleContract({
+        abi: formattedOldABI,
+      });
 
-      const tasksPaymentsContract = await TasksPayments.new();
-      await colonyResolver.register("makeTask(uint256,uint256,bytes32,uint256,uint256,uint256)", tasksPaymentsContract.address);
-      await colonyResolver.register("addPayment(uint256,uint256,address,address,uint256,uint256,uint256)", tasksPaymentsContract.address);
-      tasksPayments = await TasksPayments.at(colony.address);
+      OldInterface.setProvider(web3.currentProvider);
```

Comment:
I don't really understand what all of this stuff is. It looks like it should be encapsulated in a function kept in `scripts/deployOldUpgradeableVersion.js` and called with context-specific arguemnts.

Reply:
Yeah, good call. I'm now returning the old interface out of the setup script.